control-plane/rest/service/src/health/handlers.rs

```diff
@@ -0,0 +1,28 @@
+use crate::CachedCoreState;
+use actix_web::{get, web::Data, HttpResponse, Responder};
+
+/// Liveness probe check. Failure will result in Pod restart. 200 on success.
+#[get("/live")]
+async fn liveness(_cached_core_state: Data<CachedCoreState>) -> impl Responder {
+    HttpResponse::Ok()
+        .content_type("text/plain; charset=utf-8")
+        .insert_header(("X-Content-Type-Options", "nosniff"))
+        .body("live")
+}
+
+/// Readiness probe check. Failure will result in removal of Container from Kubernetes service
+/// target pool. 200 on success, 503 on failure.
+#[get("/ready")]
+async fn readiness(cached_core_state: Data<CachedCoreState>) -> HttpResponse {
+    if cached_core_state.get_or_assume_unavailable().await {
```

Comment:
Is this needed? What if there's an issue with the core agent, would this then result in removal of the rest?

Reply:
It's a readiness failure, that would just mean that REST won't serve requests send to the kubernetes service endpoint for mayastor-rest. Other instances (if any) of the rest would serve it. The container itself doesn't get deleted, etc. just sits around doing nothing.

Reply:
So the deletion part is done only on liveness failure?

Reply:
What error do we get if let's say core-agent is down, i.e readiness failure?

Reply:
The `1/1` on the `kubectl get po -n mayastor` turns to `0/1`. That's all. We might see some logs on the kubelet. I'm going to have to check for them. I'll post here if I manage to collect them.